src/test/codegen/likely.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// compile-flags: -C no-prepopulate-passes
+
+#![crate_type = "lib"]
+#![feature(core_intrinsics)]
+
+use std::intrinsics::{likely,unlikely};
+
+#[no_mangle]
+pub fn check_likely(x: i32, y: i32) -> Option<i32> {
+    unsafe {
+        // CHECK: call i1 @llvm.expect.i1(i1 %{{.*}}, i1 true)
```

Comment:
Could you check some surroundings around this intrinsic too? I.e. that `icmp` or something goes before the intrinsic and that branch goes right after the intrinsic?

It would look something along the lines of

```
CHECK: [[cond:%[0-9]+]] = icmp ...
CHECK-NEXT: call i1 @llvm.expect.i1(i1 [[cond]], i1 true)
CHECK-NEXT: br [[cond]] ...
```